util/cli/cli.go

```diff
@@ -57,22 +57,53 @@ func AddKubectlFlagsToCmd(cmd *cobra.Command) clientcmd.ClientConfig {
 
 // PromptCredentials is a helper to prompt the user for a username and password
 func PromptCredentials(username, password string) (string, string) {
-	for username == "" {
+	return PromptUsername(username), PromptPassword(password)
+}
+
+// PromptUsername prompts the user for a username value
+func PromptUsername(username string) string {
+	return PromptMessage("Username", username)
+}
+
+// PromptMessage prompts the user for a value
+func PromptMessage(message, value string) string {
+	for value == "" {
 		reader := bufio.NewReader(os.Stdin)
-		fmt.Print("Username: ")
-		usernameRaw, err := reader.ReadString('\n')
+		fmt.Print(message + ": ")
+		valueRaw, err := reader.ReadString('\n')
 		errors.CheckError(err)
-		username = strings.TrimSpace(usernameRaw)
+		value = strings.TrimSpace(valueRaw)
 	}
+	return value
+}
+
+func PromptPassword(password string) string {
 	for password == "" {
 		fmt.Print("Password: ")
 		passwordRaw, err := terminal.ReadPassword(syscall.Stdin)
 		errors.CheckError(err)
 		password = string(passwordRaw)
-		if password == "" {
-			fmt.Print("\n")
-		}
+		fmt.Print("\n")
+	}
+	return password
+}
+
+func AskToProceed(message string) {
```

Comment:
Wondering if a `switch` statement plus an `if` might be a little clearer than the `map[string]bool` and `for`.  Separately, wondering if maybe rather than `os.Exit`, returning a boolean would make this more flexible.

Reply:
Also a comment on this one, since it's an exported utility function, might be good for documentation purposes.

Reply:
These are great suggestions. Thanks.